### ISSUE TITLE
10$: warn on using . after ?.

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#1128]
```javascript
a?.b.c
```
Accessing `.c` might fail because `a?.b` might return `undefined`. (That's the point of `?.`, after all.)

**Comment by pepplejoshua** [comment]
I claim this for-hire task. I expect payment after I complete this task. I will email the quick-lint-js team if I am assigned this task.

**Comment by strager** [comment]
I realized that `x?.y[z]` and `x?.y()` and x?.y\`hello\` have the same issue. (Maybe more syntaxes have the issue too.) After diagnosing `x?.y.z`, we could diagnose the others with (hopefully) the same logic.

**Comment by pepplejoshua** [comment]
I will look into it. I have fixed the comments on the PR so I will look at how to get this done.
